### DISCUSSION TITLE
ThreadPoolBulkhead and Synchronous Requests

### DISCUSSION THREAD

**Discussion by DrewM15** [resilience4j/resilience4j#2032] [Q&A] [answered]
Hi!
I am attempting to transition from Spring 2.7.12 to Spring 3.1
As part of this transition, I am removing Hystrix references and replacing them with Resilience4j.
I have found a way to replicate the CircuitBreaker of Hystrix in R4J, but am having trouble with the ThreadPoolBulkhead.
From what I've found, it seems like Hystrix supports sync calls when setting up thread pools.
However I haven't found any docs supporting that ThreadPoolBulkhead supports sync calls
I ask this because I found that TimeLimiter is strictly for Async calls, and want to know if ThreadPoolBulkhead has this same limitation.
For example in the Getting Started section of the R4J docs I see:
```
The Spring Boot starter provides annotations and AOP Aspects which are auto-configured.
RateLimiter, Retry, CircuitBreaker and Bulkhead annotations support synchronous return types and asynchronous types like CompletableFuture...
```
But then the example below it shows ThreadPool bulkhead being used in an async manner
```
@Bulkhead(name = BACKEND, type = Bulkhead.Type.THREADPOOL)
public CompletableFuture<String> doSomethingAsync() throws InterruptedException {
        Thread.sleep(500);
        return CompletableFuture.completedFuture("Test");
 }
```
Does anyone know whether ThreadPoolBulkhead supports usage in sync calls?

And as a second question, is the Bulkhead type of THREADPOOL equivalent to ThreadPoolBulkhead?
Thanks!

**Comment by RobWin** [comment, marked as answer]
Hi,

1) Resilience4j does not support to timelimit sync calls. 
2) Yes

**Reply by DrewM15** [reply]
Great Thanks Rob.
Good to know ThreadPoolBulkhead only supports async.
